### ISSUE TITLE
Redo the Canvas file picker API communication

### ISSUE THREAD

**Issue by robertknight** [hypothesis/lms#353]
Re-work the Canvas file picker implementation as described in 
[this comment](https://github.com/hypothesis/lms/pull/337#issuecomment-441227838).

The TL;DR is to replace the generic Canvas API proxy in Python + JS with a route which specifically fetches a list of files associated with the current course from the LMS app.

This will make it easier to implement the following issues:
- https://github.com/hypothesis/lms/issues/257
- https://github.com/hypothesis/lms/issues/181
- https://github.com/hypothesis/lms/issues/256

An orthogonal task which will also help, in particular with https://github.com/hypothesis/lms/issues/181, is to replace the homebrew UI component code with a widely used web UI library which will reduce the amount of code we need to manage and make it easier to focus on improving the UX. Several devs at H have used React before, I think that would be fine. That should be tracked as a separate issue.

**Comment by robertknight** [comment]
This has been replaced by the more specific issues #500 and #502.
